src/web/app/pages-session/session-submission-page/session-submission-page.component.ts

```diff
@@ -415,12 +408,28 @@ export class SessionSubmissionPageComponent implements OnInit, AfterViewInit {
       }, (resp: ErrorMessageOutput) => {
         if (resp.status === 404) {
           this.simpleModalService.openInformationModal('Feedback Session Does Not Exist!', SimpleModalType.DANGER,
-              'The session does not exist (most likely deleted by the instructor after the submission link was sent).');
-          this.navigationService.navigateByURL(this.router, `/web/${this.entityType}/home`);
+              'The session does not exist (most likely deleted by the instructor after the submission link was sent).',
+              {
+                onClosed: () => this.navigationService.navigateByURL(this.router,
+                    this.loggedInUser ? `/web/${this.entityType}/home` : '/web/front/home'),
+              },
+              { backdrop: 'static' });
         } else if (resp.status === 403) {
-          this.simpleModalService.openInformationModal('Not Authorised To Access!', SimpleModalType.DANGER,
-              resp.error.message);
-          this.navigationService.navigateByURL(this.router, `/web/${this.entityType}/home`);
+          if (auth && resp.error.message === 'Login is required to access this feedback session') {
```

Comment:
I'm quite wary of using message as the condition. If missing logged in user is the condition, probably you can use `!auth.user` or `!this.loggedInUser` instead?

Reply:
Hm I see. At first, I had doubts as there were multiple reasons that cause a `403` exception, e.g. many `UnauthorizedAccessException`s in `GetFeedbackSessionAction`, but I guess it is possible in this case since `this.loadFeedbackSession()` is only called with the `auth` parameter in the case where a not logged-in user clicks on a submission link associated with a registered user, i.e. `resp.isAllowedAccess=false` and `resp.isValid=true`, is that right?